### PR TITLE
[MDS-5093] setup documentation + remove testing from husky hooks

### PR DESCRIPTION
**Pull request by taraepp**
## Objective 
Highlighting issues with current documentation:
- discoverability of documentation files
   - duplicate, outdated, or empty documentation
   - useful information that is difficult to find
- formatting/linting: There are some issues with hooks, and I believe that some of our automatic formatting broke some of the markdown in these files
- adding clarification (such as where/how to run commands)
- some minor grammatical changes
- more details in commit messages (committed each file separately)

I did up a quick organization of all the documentation files, but I'm sure it could be improved.

[MDS-5093](https://bcmines.atlassian.net/browse/MDS-5093)
[MDS-5124](https://bcmines.atlassian.net/browse/MDS-5124)

_Why are you making this change? Provide a short explanation and/or screenshots_

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,14 +21,61 @@ The product is anticipated to include:
...
+[Core API (Shared Backend)](services/core-api/README.md)
+
+- [Core API JWT](services/core-api/app/flask_jwt_oidc_local/README.md)
+  [Database](services/database/README.md)
+- [Database Migrations](migrations/README.md)
+  [Document Manager](services/document-manager/backend/README.md)
+  - [Document Manager Migrations: suggest deleting empty file](services/document-manager/backend/migrations/README.md)
+    [Fider](services/fider/README.md)
+    [Filesystem Provider](services/filesystem-provider/ej2-amazon-s3-aspcore-file-provider/README.md)
+
```
Comment (taraepp) [inline README.md]: I forgot celery! Which doesn't have a README (yet...), but it should probably be mentioned at least.

File: USAGE.md [hunk 2]
```diff
@@ -111,10 +115,10 @@ If you are on a windows machine ensure that you open powershell in administrator
 
 This will add an entry for the document manager backend if it does not currently exist.
 
-On a mac or linux run the following:
+On a mac or linux run the following in the bin directory:
 
 ```
-./AddHosts.sh add 127.0.0.1 document_manager_backend
+sudo ./AddHosts.sh add 127.0.0.1 document_manager_backend
```
Comment (taraepp) [inline USAGE.md:126]: add this to the make commands:
make docman_hosts_windows / docman_hosts_unix
eventually detect the user's OS and use the correct command --> combine into one

File: USAGE.md [hunk 3]
```diff
@@ -19,17 +19,17 @@ This file describes how to run the project and develop against it.
   - If containers are not working, they may not be enabled, enabling them in docker settings and restarting the machine fixes this
```
Comment (taraepp) [inline USAGE.md:19]: move the setup info from monorepo.md to here, make that file be about the monorepo and the process/goals of doing that

File: docs/openshift/debugging_guide.md [hunk 4]
```diff
@@ -1,70 +1,86 @@
...
 # Openshift DeploymentConfigs
+
 The majority of Core's services are still using DeploymentConfigs and these follow a hiearchal pattern where child objects (not every object is a child) inherit configuration from the parent object, that hierchary could be described as follows:
 
 |-HorizontalPodAutoscaler
 |+DeploymentConfig
- |+ReplicationController
-  |+Pod(s)
-   |+Container(s)
+|+ReplicationController
```
Comment (taraepp) [inline docs/openshift/debugging_guide.md]: maybe ignore md files for now until we create new linting rules for them

File: docs/openshift/debugging_guide.md [hunk 5]
```diff
@@ -1,70 +1,86 @@
...
 ### Quota denials
+
 `Symptoms`: Deployments fail / hang, quota warnings in `Events` tab of the resource
 
 `Cause`: There's no resource space to allocate for the new pod(s)
 
 `Fix`: adjust resource usage of this deployment or others. You can potentially request more quota from platform services but it needs to be backed up with a sysdig dashboard report
 
 ### Pre/post Hooks failing
+
```
Comment (taraepp) [inline docs/openshift/debugging_guide.md]: @vyasworks do you know about these hooks?

File: services/core-web/README.md [hunk 6]
```diff
@@ -135,6 +136,7 @@ The linting rules are built on the Airbnb configuration. Contributors should
 ensure that they are not introducing linting errors into the codebase with
 their changes. Modern text editors, such as VS Code, will indicate errors.
 See the usage guide below for more information on the linting CLI options.
+// Generally we rely on the husky hooks to lint files at commit/push. Running npm run lint in core-web produces an error about import/no-cycle rule
```
Comment (taraepp) [inline services/core-web/README.md]: Note: let's get rid of tests as part of husky hooks (they're fine to run in PRs) (PR1), and then do the linting (PR2). Also remove md files from eslint (or fix) as part of first PR.

File: README.md [hunk 7]
```diff
@@ -21,14 +21,61 @@ The product is anticipated to include:
 3. Enhancements to the public-facing [BC Mine Information website](http://mines.nrs.gov.bc.ca/) ([github repository](https://github.com/bcgov/mem-mmti-public)) to broaden the available data and create new or improved functionality to increase usability
 4. Enhancements to the [BC Mine Information website's administrative console](https://mines.empr.gov.bc.ca/) ([github repository](https://github.com/bcgov/mem-admin)) , including work required to connect with other systems (e.g., the Natural Resource Inspection System, aka NRIS)
 
+## Services
+
+// suggestion: simple description of each, perhaps organize into categories. Sub-files can either by easier to discover at top level README or more concise by linking within
+[Common: empty file, suggest delete or add content](services/common/README.md)
```
Comment (taraepp) [inline README.md]: move monorepo content into the common readme

File: README.md [hunk 8]
```diff
@@ -21,14 +21,61 @@ The product is anticipated to include:
...
+[Getting Started](docs/devops/getting_started.md)
+[Monorepo setup](docs/frontend/monorepo.md)
+
+### OpenShift Deployment
+
+[OpenShift Debugging Guide](docs/openshift/debugging_guide.md)
+[OpenShift Caveats](docs/openshift/Openshift%20Caveats.md)
+[Terraform](terraform/README.md)
+[OpenShift Database: suggest deletion](docs/openshift/database.md)
+[OpenShift PG upgrade: suggest deletion](docs/openshift/PG_9_to_13_upgrade.md)
```
Comment (taraepp) [inline README.md]: @vyasworks any value in keeping the OpenShift Database and OpenShift PG update readme files?

File: docs/openshift/debugging_guide.md (hunk 4 above)
Comment (taraepp) [reply docs/openshift/debugging_guide.md]: Changed my mind.
Installed a md preview extension (added to extension recommendations!) and neither of these look good.

File: USAGE.md (hunk 2 above)
Comment (taraepp) [reply USAGE.md:126]: For now I did not do this because of the whole admin access required to run these, but they're easier to find now

**Review by matbusby-fw — APPROVED** [review]
Awesome!  This is great.  Now we can fill in some of the areas lacking details in the following tickets!

**Pull request merged**